src/task/camera.ts

```diff
@@ -2,8 +2,53 @@ import { type User } from '../api/auth/authorization.js';
 import { CameraModel } from '../api/db/models/Camera.js';
 import { type TaskInput } from '../api/db/models/Task.js';
 import type * as gql from '../@types/graphql.js';
+import { ProjectModel } from '../api/db/models/Project.js';
+import { DeleteImagesByFilter } from './image.js';
 
 export async function UpdateSerialNumber(task: TaskInput<gql.UpdateCameraSerialNumberInput>) {
   const context = { user: { is_superuser: true, curr_project: task.projectId } as User };
   return await CameraModel.updateSerialNumber(task.config, context);
 }
+
+export async function DeleteCamera(task: TaskInput<gql.DeleteCameraInput>) {
+  const context = { user: { is_superuser: true, curr_project: task.projectId } as User };
+
+  console.log('CameraModel.deleteCameraConfig - input: ', task.config);
+  try {
+    // Step 1: delete deployments from views
+    await ProjectModel.removeCameraFromViews(
+      {
+        cameraId: task.config.cameraId,
+      },
+      context,
+    );
+    // Step 2: delete camera record from project
+    await ProjectModel.deleteCameraConfig(
+      {
+        cameraId: task.config.cameraId,
+      },
+      context,
+    );
+
+    // Step3: delete images associated with this camera
+    await DeleteImagesByFilter({
+      projectId: task.projectId,
+      config: {
+        filters: {
+          cameras: [task.config.cameraId],
+        },
+      },
+      type: 'DeleteImagesByFilter',
+      user: task.user,
+    });
+    // Step 4: unregister camera
```

Comment:
Just double-checking that this accounts for the fact that a Project can register a camera, later on unregister it, and a different Project can then register that same camera down the road. This allows a camera to be moved around to different preserves/projects etc.

At a glance I think `getWirelessCameras()` should return only the wireless cameras that _at some point_ had been registered to the current project, so in theory this condition will be met even if the wireless camera in question is now registered to a different project. 

I think that might(?) cause an issue in the `unregisterCamera()` step, because `unregisterCamera()` throws an error if the cam is not actively registered to the current project. 

I haven't tested this or dug too deep, but do you mind taking a look and testing?

Reply:
Yea you're right, the `unregisterCamera` method only works on active registrations. Will write another method for this

Reply:
I guess how should the `projRegistration` be handled? Should it just be removed from the list of registrations and if it is the current active project, create a new one for `default_project`?

Reply:
Yeah I think that's right. Find the `wirelesscamera`, if the current project is not the `active` one, just remove it from the `projRegistrations` array, but if it is active, we'll want to both remove it and set/create a `projResistration` for the `default_project`:

```javascript
// setting default project registration to active, copied from `CameraModel.unregisterCamera()` method:
const defaultProjReg = cam.projRegistrations.find((pr) => pr.projectId === 'default_project');
if (defaultProjReg) defaultProjReg.active = true;
else {
  cam.projRegistrations.push({
    _id: new ObjectId(),
    projectId: 'default_project',
    active: true,
  });
}
await cam.save();
```